### DISCUSSION TITLE
HTML Template for Weasyprint engine

### DISCUSSION THREAD

**Discussion by bmweiner** [jgm/pandoc#9741] [Q&A] [answered]
I am producing a pdf with the following pertinent options/metadata:

- pdf-engine: weasyprint
- standalone: true
- document-css: false

However, the intermediate HTML always includes some CSS styling from Pandoc. Unfortunately two lines in this styling throw warnings in weasyprint:

`gap: min(4vw, 1.5em)` (invalid value)
`overflow-x: auto` (unknown property)

How do you advise resolving these warnings? Assuming this syntax is valid, I would favor either: 1. Pandoc removes all css when setting document-css: False (seems this should be the expected outcome) or 2. Contact the weasyprint developers to understand why this syntax is not supported.

**Comment by jgm** [comment, marked as answer]
`document-css` This is not a variable or metadata field that plays any role in pandoc.

The solution is to use a custom template.  Get a copy of the default HTML5 template, and then delete the line
```
$styles.html()$
```
or substitute whatever CSS you want there.  Then use this template with `--template`.

**Reply by bmweiner** [reply]
Thanks this seems reasonable. Is it intentional that the html variable: document-css does not disable the entire partial when being set to false?

**Reply by jgm** [reply]
Hm, I hadn't remembered that the entire thing is wrapped in `if(document-css)`.  Well, yes, setting this to false should remove all of the CSS.

How are you attempting to set this variable?  Maybe you are not actually setting it to a boolean false value?

The following should work in metadata:
```
css: []
```

**Reply by bmweiner** [reply]
It doesn’t appear to me that the entire thing is wrapped. When I examine the default html template css partial with `pandoc —print-default-data-file=templates/styles.html` it looks like the `if(document-css)` is closed by an `$endif$` on line 173. I think this is just misplaced, hence why I’m seeing the extra styles in my output.

I wanted to avoid using my own template in case you all add any features I want to use in the default one in the future. 

I’m setting the `document-css` in the metadata-files list in a defaults file. I’m sure that’s working cause it removes the other css content 😊

**Reply by jgm** [reply]
You're right.  The endif is right before
```
code{white-space: pre-wrap;}
```

I think the idea was to put the conditional around "styling" CSS that might be a matter of taste, and not around CSS that really needs to be there for things to appear correctly at all.  In this case, the CSS causes divs with class `column` / `columns` to produce columnar output.  So I don't think we want to just move the `endif` forward.

I'm not sure why weasyprint has trouble with these features, which are valid. But if it's just throwing warnings, and otherwise producing good output, I'd say we should just not worry about it.

**Reply by arikrupnik** [reply]
> So I don't think we want to just move the `endif` forward.

Did you ever make this change? From a casual look at https://github.com/jgm/pandoc/blob/main/data/templates/styles.html, it appears that lines 187 through 199 still render unconditionally.

**Reply by jgm** [reply]
No, I didn't make the change, for the reason given above.
